src/lib/operators/table_scan/column_between_table_scan_impl.cpp

```diff
@@ -132,4 +140,51 @@ void ColumnBetweenTableScanImpl::_scan_dictionary_segment(const BaseDictionarySe
   });
 }
 
+void ColumnBetweenTableScanImpl::_scan_sorted_segment(const BaseSegment& segment, const ChunkID chunk_id,
+                                                      PosList& matches,
+                                                      const std::shared_ptr<const PosList>& position_filter,
+                                                      const OrderByMode order_by_mode) const {
+  resolve_data_and_segment_type(segment, [&](const auto type, const auto& typed_segment) {
+    using ColumnDataType = typename decltype(type)::type;
+
+    if constexpr (std::is_same_v<std::decay_t<decltype(typed_segment)>, ReferenceSegment>) {
+      Fail("Expected ReferenceSegments to be handled before calling this method");
+    } else {
+      auto segment_iterable = create_iterable_from_segment(typed_segment);
+      segment_iterable.with_iterators(position_filter, [&](auto segment_begin, auto segment_end) {
+        auto sorted_segment_search =
+            SortedSegmentBetweenSearch(segment_begin, segment_end, order_by_mode, predicate_condition,
+                                       boost::get<ColumnDataType>(left_value), boost::get<ColumnDataType>(right_value));
+
+        sorted_segment_search.scan_sorted_segment([&](auto begin, auto end) {
+          size_t output_idx = matches.size();
+
```

Comment:
Can [this](https://github.com/hyrise/hyrise/pull/1951) happen here?

Reply:
https://github.com/hyrise/hyrise/pull/1961/files#diff-10c2e575a4fe7a7d31742cce4bea1710R176 prevents this

Reply:
Ah nice, I see.

Another thing: It won’t make much of a difference but we could store the result of the distance calculation in a variable to determine it only once, right? Then, we could check it before we have the resize call even though it will do nothing in that case anyway. However, I doubt that there is any measurable impact :)

Reply:
Updated in 34e0d18e5a8e9f39ea1e0ff0bb33dc4e9397198b